### PR TITLE
update mwc-slider to MDC 4

### LINKED ISSUE #486: Update mwc-slider for MDC Web 4
Part of https://github.com/material-components/material-components-web-components/issues/480

mwc-slider has breaking changes in MDC web 4:

> slider: remove adapter methods appendTrackMarkers, removeTrackMarkers , setLastTrackMarkersStyleProperty , and add adapter method setTrackMarkers.

https://github.com/material-components/material-components-web-components/pulls?q=is%3Apr+is%3Aclosed deleted the 3 removed methods, and added a no-op for the new `setTrackMarkers` method, just to make tsc happy, but we presumably need to actually do something for `setTrackMarkers`.

### PR DESCRIPTION
**Pull request by e111077**
fixes #486.

Also cleaned up innerText implementation and cleaned up render function to look less ugly.

### CODE REVIEW

File: packages/slider/src/mwc-slider-base.ts [hunk 1]
```diff
@@ -174,10 +170,21 @@ export class SliderBase extends FormElement {
...
+      setTrackMarkers: (step, max, min) => {
+        const stepStr = step.toLocaleString();
+        const maxStr = max.toLocaleString();
+        const minStr = min.toLocaleString();
+        // keep calculation in css for better rounding/subpixel behavior
+        const markerAmount = `((${maxStr} - ${minStr}) / ${stepStr})`;
+        const markerWidth = '2px';
+        const markerBkgdImage = `linear-gradient(to right, currentColor ${
+            markerWidth}, transparent 0)`;
+        const markerBkgdLayout = `0 center / calc((100% - ${markerWidth}) / ${
```
Comment (azakus) [inline packages/slider/src/mwc-slider-base.ts]: This is starting to get a little complicated. Can you put a comment in here about what this calculation is doing?

Comment (e111077) [reply packages/slider/src/mwc-slider-base.ts]: Sure, I hate it, but it's exactly what they do in the component. It basically draws the notches on the slider if you enable notches and steps. It's more performant, but ugly

Comment (e111077) [reply packages/slider/src/mwc-slider-base.ts]: done

**Comment by e111077** [comment]
Re-requesting review as I refactored it a bit to not use `element.style.setProperty` to use `styleMap`. Additionally, added `{passive: true}` to some things that were giving us warnings

File: packages/slider/src/mwc-slider-base.ts [hunk 2]
```diff
@@ -137,14 +146,18 @@ export class SliderBase extends FormElement {
       computeBoundingRect: () => this.mdcRoot.getBoundingClientRect(),
       getTabIndex: () => this.mdcRoot.tabIndex,
       registerInteractionHandler:
-          <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
-              this.mdcRoot.addEventListener(type, handler),
+          <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
+            const init = type === 'touchstart' ? {passive: true} : {};
```
Comment (azakus) [inline packages/slider/src/mwc-slider-base.ts]: This should be using `applyPassive()` from `@material/dom/events` like we use in ripple-directive: https://github.com/material-components/material-components-web-components/blob/master/packages/ripple/src/ripple-directive.ts#L17

File: packages/slider/src/mwc-slider-base.ts [hunk 3]
```diff
@@ -137,14 +146,18 @@ export class SliderBase extends FormElement {
...
+            this.mdcRoot.addEventListener(type, handler, init);
+          },
       deregisterInteractionHandler:
           <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
               this.mdcRoot.removeEventListener(type, handler),
       registerThumbContainerInteractionHandler:
-          <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
-              this.thumbContainer.addEventListener(type, handler),
+          <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
+            const init = type === 'touchstart' ? {passive: true} : {};
```
Comment (azakus) [inline packages/slider/src/mwc-slider-base.ts]: `applyPassive()`

File: packages/slider/src/mwc-slider-base.ts [hunk 4]
```diff
@@ -155,7 +168,7 @@ export class SliderBase extends FormElement {
           <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
               document.body.removeEventListener(type, handler),
       registerResizeHandler: (handler: SpecificEventListener<'resize'>) =>
-          window.addEventListener('resize', handler),
+          window.addEventListener('resize', handler, {passive: true}),
```
Comment (azakus) [inline packages/slider/src/mwc-slider-base.ts]: `applyPassive()`

File: packages/slider/src/mwc-slider-base.ts [hunk 5]
```diff
@@ -127,6 +131,11 @@ export class SliderBase extends FormElement {
     </div>`;
   }
 
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.mdcFoundation.destroy();
```
Comment (azakus) [inline packages/slider/src/mwc-slider-base.ts]: Why destroy the foundation on `disconnectedCallback()`?

Comment (e111077) [reply packages/slider/src/mwc-slider-base.ts]: to remove all the body resize observers

File: packages/slider/src/mwc-slider-base.ts [hunk 6]
```diff
@@ -147,15 +148,15 @@ export class SliderBase extends FormElement {
       getTabIndex: () => this.mdcRoot.tabIndex,
       registerInteractionHandler:
           <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
-            const init = type === 'touchstart' ? {passive: true} : {};
+            const init = type === 'touchstart' ? applyPassive() : {};
```
Comment (azakus) [inline packages/slider/src/mwc-slider-base.ts]: The fallback must be `undefined` to not force `capture` behavior on IE

File: packages/slider/src/mwc-slider-base.ts [hunk 7]
```diff
@@ -147,15 +148,15 @@ export class SliderBase extends FormElement {
...
             this.mdcRoot.addEventListener(type, handler, init);
           },
       deregisterInteractionHandler:
           <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
               this.mdcRoot.removeEventListener(type, handler),
       registerThumbContainerInteractionHandler:
           <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
-            const init = type === 'touchstart' ? {passive: true} : {};
+            const init = type === 'touchstart' ? applyPassive() : {};
```
Comment (azakus) [inline packages/slider/src/mwc-slider-base.ts]: fallback to `undefined`

File: packages/slider/src/mwc-slider-base.ts (hunk 5 above)
Comment (azakus) [reply packages/slider/src/mwc-slider-base.ts]: Seems like we either need an explicit removeEventListener wrapper in the foundation, or we'll need to recreate the foundation in `connectedCallback()`

Comment (e111077) [reply packages/slider/src/mwc-slider-base.ts]: re-initted in connected callback

**Pull request merged**